coordinator/src/orderbook/routes.rs

```diff
@@ -147,7 +147,7 @@ pub async fn post_order(
     let matched_orders = match_order(order.clone(), all_non_expired_orders)
         .map_err(|e| AppError::InternalServerError(format!("Failed to match order: {e:#}")))?;
 
-    let authenticated_users = state.authenticated_users.lock().await;
+    let authenticated_users = state.authenticated_users.lock().await.clone();
```

Comment:
❓ I assume that the `clone` will free the lock immediately, so we are now only locking for the duration of the clone.
Is my understanding correct?

Reply:
Yes, exactly. Before we have held the lock until the variable went out of scope, ie. until the end of the function